requester.go

```diff
@@ -0,0 +1,40 @@
+package http
+
+import (
+	"context"
+
+	"github.com/blugnu/errorcontext"
+	"github.com/blugnu/http/request"
+)
+
+type Returning[T any] struct {
+	HttpClient
+}
+
+func (r Returning[T]) Get(ctx context.Context, path string, opts ...func(*Request) error) (*T, error) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/blugnu/http/pull/3#issuecomment-2211525124>#3867f3</a><span> - 07/06/2024, 12:12 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'Get' method in the 'Returning' struct does not handle HTTP status codes that indicate an error (e.g., 4xx or 5xx status codes). This can lead to incorrect assumptions about the success of the request.
</div>
<div id="fix">
<b>Fix:</b> Check the HTTP status code of the response and handle error status codes appropriately.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -16,6 +16,11 @@ func (r Returning[T]) Get(ctx context.Context, path string, opts ...func(*Request) error) (*T, error) {
  	resp, err := r.HttpClient.Get(ctx, path, opts...)
  	if err != nil {
  		return nil, errorcontext.Wrap(ctx, err)
 +	}
 +	if resp.StatusCode >= 400 {
 +		return nil, errorcontext.Errorf(ctx, "HTTP request failed with status code %d", resp.StatusCode)
 +	}
 +	defer resp.Body.Close()
 
  	result, err := UnmarshalJSON[T](resp)
  	if err != nil {
```
</div>
</details>
</div>
<br>


</div>